### PR TITLE
Feat/29 Create a Member Card Component for Dashboard

### LINKED ISSUE #14: [FEAT] Create a Member Card Component for Dashboard
Develop a card component to display the members of the organization on the dashboard. This component should include the profile image, name, email address, a dropdown for roles, and additional information icon. The admin should be able to change roles and access other member-specific functionalities.

### Purpose
This component will allow the admin to properly visualize the members of the organization and manage their roles and other functionalities efficiently.

### Acceptance Criteria

**Requirements**
The component should receive the following necessary props

- Profile Image - The image of the user must be displayed
- Full Name
- Email Address
- Role dropdown - A dropdown displaying the roles.
- Role Change Action - A function to handle role changes by receiving the id of the user
- Other Info Dropdown -  A dropdown for additional actions 
- Delete Action - A function to delete the member by receiving the id of the user to delete

**Tasks**

- [ ]  Design the UI based on the provided design mockups.
- [ ]  Ensure the UI is responsive and adapts to different screen sizes.
- [ ]  Implement props to receive profile image, full name, email address, role dropdown, role change action, other info dropdown, and delete action.
- [ ] Implement state management to handle role changes and other member-specific actions.
- [ ] Implement role management functionality with the appropriate API endpoints to update member roles..
- [ ] Implement delete functionality for members with the appropriate API endpoints to delete members.
- [ ] Write unit tests to cover different states and props.

**Note**
- Ensure code quality and maintainability.
- Provide clear documentation and usage examples.

Deadline: Thursday 11pm

Images and Figma Link

Screenshot 2024-07-17 131702
image
image

https://www.figma.com/design/VEItfX6St5NSAqqNHImcxD/HNG-Boilerplate-Designs?node-id=96-100753&m=dev

### PR DESCRIPTION
**Pull request by Saviourise**
# Description

**Closes #14**

# Changes proposed

## What were you told to do?

Create a Member Card Component for Dashboard

## What did you do?

1. I created a member card component for dashboard
2. I added neccesary props so that the component can be fully dynamic
3. I followed design implementation and made sure the card is responsive
4. I used shadcn for select and dropdown
5. I tested the component to make sure it is working

# Check List (Check all the applicable boxes)

🚨Please review the [contribution guideline](CONTRIBUTING.md) for this repository.

- [x] My code follows the code style of this project.
- [x] This PR does not contain plagiarized content.
- [x] The title and description of the PR is clear and explains the approach.
- [x] I am making a pull request against the **dev branch** (left side).
- [x] My commit messages styles matches our requested structure.
- [x] My code additions will fail neither code linting checks nor unit test.
- [x] I am only making changes to files I was requested to.

# Screenshots/Videos

https://github.com/user-attachments/assets/8c1110f8-1095-466f-81c3-abe11d1aa53d

Screenshot 2024-07-20 170247
Screenshot 2024-07-20 171448

### CODE REVIEW

File: src/components/common/DashboardMemberCard/index.css
Comment (skeby) [inline src/components/common/DashboardMemberCard/index.css:1]: You were told not to create your own css files
Use tailwind boss

**Review by skeby — CHANGES_REQUESTED** [review]
Attach screenshots/video of the stuff you built
Ensure it shows the different states of your component

File: src/components/common/DashboardMemberCard/index.css
Comment (saviour-encentral) [reply src/components/common/DashboardMemberCard/index.css:1]: I added CSS because what I was trying to achieve was not available on tailwind @skeby

Comment (skeby) [reply src/components/common/DashboardMemberCard/index.css:1]: I don't understand?
What happened to `font-medium text-sm text-[#0f172a]`

If you want your work to be reviewed, remove this css file

Comment (saviour-encentral) [reply src/components/common/DashboardMemberCard/index.css:1]: I have removed the file @skeby , that is why I said you should review again

Comment (saviour-encentral) [reply src/components/common/DashboardMemberCard/index.css:1]: I have removed the file @skeby , that is why I said you should review again

**Pull request closed without merging**